pkg/server/server.go

```diff
@@ -69,6 +72,8 @@ type Options struct {
 	ServerVersion              string
 	// SQLCache enables the SQLite-based lasso caching mechanism
 	SQLCache bool
+
+	ExtensionAPIServer *ext.ExtensionAPIServer
```

Comment:
Initially I didn't see that steve was being "started" somewhere so I figured I'd just pass in the extensionAPIServer to steve, and steve would only use it as a `http.Handler`. For this reason, the current API looks like the following:

```go
         // Configure a bunch of options
          opts := ExtensionAPIServerOptions{                                                                                                                                                                                                                           
                  GetOpenAPIDefinitions: getOpenAPIDefinitions,                                                                                                                                                                                                        
                  OpenAPIDefinitionNameReplacements: map[string]string{                                                                                                                                                                                                
                          "com.github.rancher.steve.pkg.ext": "io.cattle.ext.v1",                                                                                                                                                                                      
                  },                                                                                                                                                                                                                                                   
          }                                                                                                                                                                                                                                                            

         // Create the extension api server                                                                                                                                                                                                                                            
          extensionAPIServer, err := NewExtensionAPIServer(scheme, codecs, opts)                                                                                                                                                                                       
          if err != nil {                                                                                                                                                                                                                                              
                  return nil, func() {}, err                                                                                                                                                                                                                           
          }                                                                                                                                                                                                                                                            
                
          // install one or more stores on it                                                                                                                                                                                                                                                       
          InstallStore(extensionAPIServer, objT, objTList, "testtypes", "testtype", testTypeGV.WithKind("TestType"), store)                                                                                                                                            
                                                                                                                                                                                                                                                                       
          ctx, cancel := context.WithCancel(context.Background())                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
          go func() {             
                // run the store                                                                                                                                                                                                                                      
                  err := extensionAPIServer.Run(ctx, readyCh)                                                                                                                                                                                                                                                                                                                                                                                                                                         
          }()                                                                                                                                                                                                                                                          

        // Steve is given the extensionAPIServer
         steve.New(...., extensionAPIServer)
```

So in that current way, steve could simply accept a `http.Handler` for the extension aPI server.


But now that I realize that steve is something that we do start, maybe instead the extension api server becomes a

```go
type HandlerRunner struct {
   http.Handler
   Run(context.Context)
}
```

and we can start the extension apiserver inside of steve, instead of outside.. I'm still not sure about what we gain immediately with making in an interface, because Steve will already accept a `nil` extensionapiserver. I see two consumers:
- Rancher: Will definitely have the extensionAPIServer set. Maybe the interface would be useful here for tests inside Rancher, although I'd rather have Rancher use the _real_ deal to be more confident. (Note: having the real extension api server being used shouldn't be a problem because _it_ accepts a bunch of interfaces that can be mocked if needed).
- Others (rancher desktop): Will have extensionAPIServer set to nil. I don't see the interface being useful here.

In any case, let me have a go at it and I'll see where this leads me.

Reply:
I think that I would overall disagree with the reasoning on the two use cases:
- Rancher.  In this case, I think that Rancher can simply do something like "steve.DefaultExtensionAPIServer". So I don't think having an interface here prevents you from using the real deal.
- Others. In this case, I think allowing people to do something simple would make sense. Ultimately, the purpose of this is to add some "Abitrary endpoints" - I think that could be useful to other use cases of steve. I also think it would be easier to understand what this is doing at a high level - instead of this massive struct that accepts stores and does complex openapi generation with Authentication and all this other stuff, it's a more simple interface that is easier to understand.

Reply:
I'm addressing this in 294188c7b4e2e67d2ecbb548716bdb5a1234eaf7.

I have defined an interface in pkg/server/server.go since that's the consumer of the extension api server. Resolving but feel free to re-open.